Fixed map finished panel not being closable when replay file can't be written

braem opened the pull request:
Closes #793 - Added the issue to the 0.8.5 release as well since it's hud related. Apologies if I shouldn't have done this.

Fixed this issue by fixing up some relevant code.

The `replay_save` event was not fired when the replay file was unable to be written to, despite it having a boolean `save` field that specifies whether it succeeded or not. This means every time `replay_save` is fired it's `save` field is `true`.
Every catching of this event checks whether `save` is `true`:
- `mom_run_poster.cpp`
- `hud_comparisons.cpp`
- `hud_mapfinished.cpp`
- `ClientTimesDisplay.cpp`

So having it fire regardless of whether saving succeeded has no repercussions.

Made the replay save button on the map finished panel disabled until the `replay_save` event is fired with a `save` field of `true`. Without this, clicking the play replay button when the replay fails to save prevents the panel from closing. This shouldn't be clickable anyways until the replay is saved.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

Gocnak commented on mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp:
```diff
@@ -156,6 +157,7 @@ void CHudMapFinishedDialog::LevelShutdown()
     m_pRunData = nullptr;
     m_bIsGhost = false;
     m_bCanClose = false;
+    m_pPlayReplayButton->SetEnabled(false);
```
Is this one necessary? Wouldn't the one in Reset catch the use case of this?

Gocnak commented on mp/src/game/server/momentum/mom_replay_system.cpp:
```diff
@@ -144,20 +144,20 @@ void CMomentumReplaySystem::FinishRecording()
 
+    if (bStoredReplay)
+    {
+        Log("Recording Stopped! Ticks: %i\n", m_pRecordingReplay->GetFrameCount());
```
While we're here can we implement the simple suggestion of also printing out the run time to console? Should just be a call to `MomUtil::FormatTime` with the recording replay's *run time* (not frame count)

braem commented:
Removed that useless call in `LevelShutdown()`

Added the time printout.
![image](https://user-images.githubusercontent.com/9014762/83373611-dbacdd00-a37d-11ea-998d-f407635500b1.png)
